tools/cli/internal/openapi/oasdiff.go

```diff
@@ -234,7 +234,9 @@ func (o OasDiff) mergeTags() error {
 			}
 		}
 	}
-	sort.Sort(ByName(baseTags))
+	slices.SortFunc(ByName(baseTags), func(a, b *openapi3.Tag) int {
+		return strings.Compare(strings.ToLower(a.Name), strings.ToLower(b.Name))
```

Comment:
# Testing

Using the foas generated by the evergreen task and run locally ` bazel run //scripts/evergreen/openapi:spectral_lint`
```bash
Linting all OpenAPI specs(OASes).
 - linting /Users/andrea.angiolillo/workspace/mms/server/openapi/atlas/openapi-foas-dev.yaml

/Users/andrea.angiolillo/workspace/mms/server/openapi/atlas/openapi-foas-dev.yaml
 46859:6  error  xgen-docs-tags-alphabetical  Tags should be defined in alphabetical order.  tags

✖ 1 problem (1 error, 0 warnings, 0 infos, 0 hints)
```


Genering the foas with the foascli binary of this change and running ` bazel run //scripts/evergreen/openapi:spectral_lint`

```bash

6.11.1
Linting all OpenAPI specs(OASes).
 - linting /Users/andrea.angiolillo/workspace/mms/server/openapi/atlas/openapi-foas.yaml
No results with a severity of 'error' found!
 - linting /Users/andrea.angiolillo/workspace/mms/server/openapi/services/openapi*.yaml
No results with a severity of 'error' found!
Finished spectral linter
For errors visit https://wiki.corp.mongodb.com/x/_4DsD
```